numpy/core/src/multiarray/descriptor.c

```diff
@@ -3814,6 +3814,106 @@ _subscript_by_index(PyArray_Descr *self, Py_ssize_t i)
     return _subscript_by_name(self, name);
 }
 
+static npy_bool
+_is_list_of_strings(PyObject *obj)
+{
+    int seqlen, i;
+    if (!PyList_Check(obj)) {
+        return NPY_FALSE;
+    }
+    seqlen = PyList_GET_SIZE(obj);
+    for (i = 0; i < seqlen; i++) {
+        PyObject *item = PyList_GET_ITEM(obj, i);
+#if defined(NPY_PY3K)
+        if (!PyUnicode_Check(item)) {
+#else
+        if (!PyString_Check(item) && !PyUnicode_Check(item)) {
+#endif
+            return NPY_FALSE;
+        }
+    }
+
+    return NPY_TRUE;
+}
+
+static PyObject *
+_subscript_by_list_of_strings(PyArray_Descr *self, PyObject *ind)
+{
+    int seqlen, i;
+    PyObject *fields;
+    PyObject *names;
```

Comment:
Initialize these to NULL so XDECREF doesn't fail in the first few goto fails.

Reply:
Good catch